app/src/main/kotlin/model/WeightedDirectedGraph.kt

```diff
@@ -78,4 +78,60 @@ class WeightedDirectedGraph<D> : DirectedGraph<D>() {
         }
         return path.reversed()
     }
+
+    // returns null if path doesn't exist
+    fun findShortestPathFordBellman(srcVertex: Vertex<D>, destVertex: Vertex<D>): List<Pair<Edge<D>, Vertex<D>>>? {
+        val NEG_INF = -1000000
+
+        val distance = MutableList(vertices.size) { Int.MAX_VALUE }
+        val predecessor = MutableList<Vertex<D>?>(vertices.size) { null }
+
+        distance[srcVertex.id] = 0
+
+        for (i in 0..vertices.size - 1) {
```

Comment:
Can change to `for (i in vertices.indices)`

Reply:
It is specific for the algo the iterate exactly number of vertices - 1 times, my code emphasizes that, but thanks for the remark!!